overlord/devicestate/firstboot20_test.go

```diff
@@ -162,7 +250,20 @@ func (s *firstBoot20Suite) testPopulateFromSeedCore20Happy(c *C, m *boot.Modeenv
 	tsAll, err := devicestate.PopulateStateFromSeedImpl(st, &opts, s.perfTimings)
 	c.Assert(err, IsNil)
 
-	checkOrder(c, tsAll, "snapd", "pc-kernel", "core20", "pc")
+	snaps := []string{"snapd", "pc-kernel", "core20", "pc"}
+	allDevModeSnaps := stripSnapNamesWithChannels(extraDevModeSnaps)
+	if len(extraDevModeSnaps) != 0 {
+		snaps = append(snaps, allDevModeSnaps...)
+	}
+	checkOrder(c, tsAll, snaps...)
+
+	// if the model is dangerous check that all snaps in the model have the flag
```

Comment:
the "all" seems wrong here given we are checking DevMode itself, not ApplySnapDevMode ?

Reply:
yes this was an old implementation in the test, I forgot to update the comment